src/Workspaces/Core/Portable/Indentation/IIndentationService.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Indentation
+{
+    /// <summary>
+    /// An indentation result represents where the indent should be placed.  It conveys this through
+    /// a pair of values.  A position in the existing document where the indent should be relative,
+    /// and the number of columns after that the indent should be placed at.  
+    /// 
+    /// This pairing provides flexibility to the implementor to compute the indentation results in
+    /// a variety of ways.  For example, one implementation may wish to express indentation of a 
+    /// newline as being four columns past the start of the first token on a previous line.  Another
+    /// may wish to simply express the indentation as an absolute amount from the start of the 
+    /// current line.  With this tuple, both forms can be expressed, and the implementor does not
+    /// have to convert from one to the other.
+    /// </summary>
+    internal struct IndentationResult
```

Comment:
I think this type should either contain sourcetext this belong to or TextLine so that we know what data is number is based off.

Reply:
since this refactoring is making document to be passed around, we might just add Document here so that snapshot is very clear